app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java

```diff
@@ -25,38 +25,100 @@
 import android.support.v7.widget.RecyclerView;
 import android.text.SpannableString;
 import android.text.style.StyleSpan;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import com.g11x.checklistapp.data.Checklist;
-import com.g11x.checklistapp.data.ChecklistManager;
+import com.g11x.checklistapp.data.ChecklistItem;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class ChecklistActivity extends AppCompatActivity {
 
-  private ChecklistAdapter adapter;
+  private ChecklistAdapter checklistAdapter;
+  private DatabaseReference checklistRef;
+  private final List<ChecklistItem> checklistItems = new ArrayList<>();
+  private ChildEventListener childEventListener;
+
+  private void addChildListeners(DatabaseReference reference) {
+    childEventListener = new ChildEventListener() {
+      @Override
+      public void onChildAdded(DataSnapshot dataSnapshot, String s) {
```

Comment:
Any thoughts on using [FirebaseUI's adapter](https://github.com/firebase/FirebaseUI-Android/blob/master/database/README.md#using-firebaseui-to-populate-a-recyclerview)?